server/src/main/java/org/elasticsearch/common/time/CompatCalendarNameProvider.java

```diff
@@ -0,0 +1,649 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.time;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.core.UpdateForV9;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.spi.CalendarNameProvider;
+
+/**
+ * Class containing the differences ONLY from CLDR that were in the old COMPAT locale set.
+ * These have been copied here to ensure date format compatibility for all v8 Elasticsearch releases.
+ * <p>
+ * The source data can be obtained by iterating through {@link Locale#getAvailableLocales()}
+ * for both {@code -Djava.locale.providers=CLDR} and {@code -Djava.locale.providers=COMPAT},
+ * obtaining {@link java.util.Calendar#getInstance} for each locale,
+ * and calling {@code java.util.Calendar#getDisplayNames} for all valid values of {@code field} and {@code style}.
+ * This data can then be written to disk in an appropriate textual format, and compared with a diff tool.
+ * <p>
+ * Locales with a country extension will just have the same english-base info as {@link Locale#ROOT}, and so can be ignored.
+ * We're only concerned about the top-level language differences here.
+ */
```

Comment:
to confirm, that class was generated based on a data described in the javadoc?

Reply:
This is not auto-generated; I generated the diffs, then created the overrides manually. I wanted to get something working quickly to validate this does actually fix the test failures, and merge it fairly quickly.

We can discuss how formal we want to make this override as part of longer-term support for this class - possibly creating gradle tasks for generating the locale files, doing the diff, creating the overrides file, etc...